cpp/src/arrow/type.cc

```diff
@@ -131,6 +133,7 @@ std::string ToString(Type::type id) {
     TO_STRING_CASE(FLOAT)
     TO_STRING_CASE(DOUBLE)
     TO_STRING_CASE(DECIMAL)
```

Comment:
Shouldn't this be changed to `DECIMAL128`?
(in general, do a search for `DECIMAL` in all the C++ code, this may catch some overloooked instances)

Reply:
this one should.  We have DECIMAL for backwards compatibility, I think the remaining places that it is used are places we will need to update to support Decimal256.  By leaving them as DECIMAL we can find them easily with by commenting out the alias.  Does this sound reasonable?

Reply:
Ok. Maybe we can deprecate the alias at some point?

Reply:
yes, that is the intent.  I'll be opening up a bunch of JIRA work to track down usages and remove.  Partial list so far:
- CSV
- Ruby/Gobj bindings
- Implementation for Parquet
- Finish Python implementation (rescaling is needed)
- Gandiva 
- Computation kernels (in particular casts)

Likely some others ...